### PR TITLE
LPS-55189

### PR DESCRIPTION
**Pull request by mhan810**

### CODE REVIEW

**Comment by mhan810** [comment]
Hey Shuyang, sending this to you since I think the reason we have a lot of test breakages is due to the fact that we aren't initializing modules when we use the test rule: LiferayIntegrationTestRule.  Seems like they are only initialized when we add MainServletTestCallback.

Is that accurate?  With message bus moving out, we'll need to init modules.

**Comment by shuyangzhou** [comment]
ci:retest

**Comment by shuyangzhou** [comment]
OSGi framework is initialized inside InitUtil.initWithSpring() which is called from LiferayIntegrationTestRule.

We should have all modules loaded, however we have to do the "ModuleFrameworkUtilAdapter.registerContext(mockServletContext);" inside MainServletTestCallback, as there is no ServletContext got created in LiferayIntegrationTestRule. So if the modules depends on the registeration of portal's ServletContext, it won't be actually initialized properly without MainServletTestCallback.

We have to keep LiferayIntegrationTestRule and MainServletTestRule separated, as for Arquillian tests, they run inside a real portal, which we can not re-publish a different servlet context, so they can not use the logic from MainServletTestRule, but they still need stuff from LiferayIntegrationTestRule.

CI is currently breaking, QA made something yesterday caused this, they are working it now. When it gets back up running, I will check the errors, See what's the best way to make this work.

**Comment by shuyangzhou** [comment]
ci:retest

**Comment by shuyangzhou** [comment]
@mhan810 Look at this failure https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-backend%28master%29/group=0,label_exp=!test-1-2/589/consoleFull

I am pretty sure this is a real bug.

The actual failing test log is :

    [junit] Testcase: com.liferay.portal.kernel.lar.lifecycle.ExportImportLifecycleEventListenerRegistryUtilTest:	Caused an ERROR
    [junit] org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'messagingConfigurator.core' defined in class path resource [META-INF/messaging-core-spring.xml]: Invocation of init method failed; nested exception is java.lang.NullPointerException
    [junit] java.lang.RuntimeException: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'messagingConfigurator.core' defined in class path resource [META-INF/messaging-core-spring.xml]: Invocation of init method failed; nested exception is java.lang.NullPointerException
    [junit] 	at com.liferay.portal.util.InitUtil.initWithSpring(InitUtil.java:223)
    [junit] 	at com.liferay.portal.test.rule.LiferayIntegrationTestRule.apply(LiferayIntegrationTestRule.java:53)
    [junit] 	at com.liferay.portal.kernel.test.rule.AggregateTestRule.apply(AggregateTestRule.java:55)
    [junit] Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'messagingConfigurator.core' defined in class path resource [META-INF/messaging-core-spring.xml]: Invocation of init method failed; nested exception is java.lang.NullPointerException
    [junit] 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1513)
    [junit] 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:521)
    [junit] 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:458)
    [junit] 	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:293)
    [junit] 	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:223)
    [junit] 	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:290)
    [junit] 	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:191)
    [junit] 	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:633)
    [junit] 	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:932)
    [junit] 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:479)
    [junit] 	at org.springframework.context.support.ClassPathXmlApplicationContext.<init>(ClassPathXmlApplicationContext.java:139)
    [junit] 	at org.springframework.context.support.ClassPathXmlApplicationContext.<init>(ClassPathXmlApplicationContext.java:105)
    [junit] 	at com.liferay.portal.util.InitUtil.initWithSpring(InitUtil.java:207)
    [junit] Caused by: java.lang.NullPointerException
    [junit] 	at com.liferay.portal.kernel.messaging.config.AbstractMessagingConfigurator.afterPropertiesSet(AbstractMessagingConfigurator.java:72)
    [junit] 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeCustomInitMethod(AbstractAutowireCapableBeanFactory.java:1639)
    [junit] 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1580)
    [junit] 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1510)
    [junit] 
    [junit] 
    [junit] TEST com.liferay.portal.kernel.lar.lifecycle.ExportImportLifecycleEventListenerRegistryUtilTest FAILED

And I can debug to reproduce it.

Basically, the DefaultMessageBus is being published by "fileinstall-/opt/liferay-portal-trunk/osgi/portal" thread. And the Spring context (including messaging-core-spring.xml) is initialized by "main" thread. So they are racing each other. And we have a good chance to see a null from main thread. 

So let's do same old sleep waiting here?

**Comment by mhan810** [comment]
Ugh k I thought the MB was initializing fast enough since it doesn't really
have any other dependencies.  Will dig thru this later when I get back.
